lib/pf/pfcmd/checkup.pm

```diff
@@ -965,23 +965,27 @@ Make sure only one external authentication source is selected for each type.
 # TODO: We might want to check if specified auth module(s) are valid... to do so, we'll have to separate the auth thing from the extension check.
 sub portal_profiles {
 
-    my $profile_params = qr/(?:locale|filter|logo|guest_self_reg|guest_modes|template_path|billing_engine|description|sources|redirecturl|always_use_redirecturl|mandatory_fields|nbregpages|allowed_devices|allow_android_devices|reuse_dot1x_credentials|provisioners)/;
+    my $profile_params = qr/(?:locale|filter|logo|guest_self_reg|guest_modes|template_path|billing_engine|description|sources|redirecturl|always_use_redirecturl|mandatory_fields|nbregpages|allowed_devices|allow_android_devices|reuse_dot1x_credentials|provisioners|filter_match_style)/;
 
     foreach my $portal_profile ( $cached_profiles_config->Sections) {
-        if ($portal_profile ne 'default' && !-d "$install_dir/html/captive-portal/profile-templates/$portal_profile") {
+        my $data = $Profiles_Config{$portal_profile};
+        # Checks for the non default profiles
+        if ($portal_profile ne 'default' ) {
             add_problem( $WARN, "template directory '$install_dir/html/captive-portal/profile-templates/$portal_profile' for profile $portal_profile does not exist using default templates" )
+                if (!-d "$install_dir/html/captive-portal/profile-templates/$portal_profile");
+
+            add_problem ( $FATAL, "missing filter parameter for profile $portal_profile" )
+                if (!defined($data->{'filter'}) );
         }
 
-        add_problem ( $FATAL, "missing filter parameter for profile $portal_profile" )
-            if ( $portal_profile ne 'default' &&  !defined($Profiles_Config{$portal_profile}{'filter'}) );
 
-        foreach my $key ( keys %{$Profiles_Config{$portal_profile}} ) {
+        foreach my $key ( keys %$data ) {
             add_problem( $WARN, "invalid parameter $key for profile $portal_profile" )
                 if ( $key !~ /$profile_params/ );
         }
 
         my %external;
-        foreach my $source ( grep { $_ && $_->class eq 'external' } map { pf::authentication::getAuthenticationSource($_) } @{$Profiles_Config{$portal_profile}{'sources'}} ) {
+        foreach my $source ( grep { $_ && $_->class eq 'external' } map { pf::authentication::getAuthenticationSource($_) } @{$data->{'sources'}} ) {
```

Comment:
I think that a condition like this one would require just a single line of comment from the developper.
It will save someone else some time to understand what we're dealing with and that single line of comment only takes 10 seconds to write.